tests/schema/test_arguments.py

```diff
@@ -105,6 +105,29 @@ def hello(self, name: Optional[str] = UNSET, age: Optional[int] = UNSET) -> str:
     assert result.data == {"hello": "Hi there"}
 
 
+def test_optional_argument_unset_with_exhaustive_typing():
```

Comment:
Can these test be narrowed in scope and moved to be with the other type resolving tests for annotations? I don't think we need to going all the way to query execution to test this.

Reply:
Great call out!!

I choose this test since it's the closest to an integration test that ensures the annotation parser works and it correctly still considers the schema output to be a nullable field.

I've updated the tests in `tests/types/resolving/test_optionals.py` as well. Should I remove the ones in `test_arguments`? Or would it be useful to keep them in here as well?

Reply:
@BryceBeagle can you take another 👀  at this PR?

Reply:
With the tests in tests/types/resolving I think these tests here are redundant. No codepath should escape those tests and make it here undetected.

Reply:
Updated!